atoma-bin/atoma_node.rs

```diff
@@ -153,6 +153,24 @@ async fn initialize_tokenizers(
     try_join_all(fetch_futures).await
 }
 
+fn spawn_with_shutdown<F, E>(
```

Comment:
1. Add docs.
2. This method is repeated, we could add a utils crate for this sort of functionality, but it can be done on a posterior iteration